src/Forms/CurrencyField.php

```diff
@@ -54,21 +55,22 @@ public function performReadonlyTransformation()
         return $this->castedCopy(CurrencyField_Readonly::class);
     }
 
-    public function validate($validator)
+    public function validate(): ValidationResult
     {
-        $result = true;
-        $currencySymbol = preg_quote(DBCurrency::config()->uninherited('currency_symbol') ?? '');
-        $regex = '/^\s*(\-?' . $currencySymbol . '?|' . $currencySymbol . '\-?)?(\d{1,3}(\,\d{3})*|(\d+))(\.\d{2})?\s*$/';
-        if (!empty($this->value) && !preg_match($regex ?? '', $this->value ?? '')) {
-            $validator->validationError(
-                $this->name,
-                _t('SilverStripe\\Forms\\Form.VALIDCURRENCY', "Please enter a valid currency"),
-                "validation"
-            );
-            $result = false;
-        }
-
-        return $this->extendValidationResult($result, $validator);
+        $result = ValidationResult::create();
+        $this->beforeExtending('updateValidate', function () use ($result) {
```

Comment:
```suggestion
        $this->beforeExtending('updateValidate', function (ValidationResult $result) {
```
We should be updating the result that's passed into the extension hook.

Reply:
Updated